### ISSUE TITLE
[SC Testing]: Missing legend element on contact information page

### ISSUE THREAD

**Issue by davidakennedy** [department-of-veterans-affairs/va.gov-team#92621]
## What happened?

**Describe what went wrong**:

When you hit the "Contact information" page on the SC test form `https://staging.va.gov/decision-reviews/test-sc/contact-information`, there isn't a legend around the h3 heading. That means that screen readers/assistive tech may not hear the contact information heading when navigating to a control inside the fieldset.

**Add the steps to reproduce the bug**:

1. Navigate to `https://staging.va.gov/decision-reviews/test-sc/contact-information` 
2. Open dev tools, and not missing legend around h3 heading.

**Add a screen recording or photo**:

<img width="546" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOVhGQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--26730b95c1712e22f02e91517e46db29bd63aaca/Screenshot%202024-09-10%20at%205.04.03%E2%80%AFPM.png" alt="Contact information page with red arrow pointing to heading with missing legend element." />

## How severe is the problem?

- [ ] **Critical**: User is block or significantly hindered from submitting the form, data is lost, or claim and evidence test coverage is missing. Fix ASAP.
- [x] **Moderate**: User is not blocked, but the happy path is interrupted. Fix after critical bugs.
- [ ] **Minor**: Cosmetic bugs and minor language inconsistencies. Defer to a later sprint.

## What is the desired behavior?

1. _When a user does X, the **visual** result is y._
2. _When a user does X, the **backend** result is y._
3. _When a user does X, the **event logging** result is y._

When the Veteran navigates to controls on the page, in the fieldset, they should hear the "Contact information" announced.

## Definition of Done

- [ ] Acceptance criteria reviewed and approved by product or design
- [ ] Unit test (or other test coverage) added if possible
- [ ] Logging and monitoring in place

**Comment by davidakennedy** [comment]
HI @Mottie! I gave this a look after your changes, and everything looks good. Removing the `<fieldset>` matches production, so that's great.
